zfs-config.sh

```
@@ -129,29 +137,108 @@ stopped_containers=()
+$current_user ALL=(ALL) NOPASSWD: /usr/bin/apt install -y jq
+$current_user ALL=(ALL) NOPASSWD: /usr/bin/apt install -y sanoid
+
+# File system operations that the scripts need
+$current_user ALL=(ALL) NOPASSWD: /bin/mkdir -p *
+$current_user ALL=(ALL) NOPASSWD: /bin/cp /etc/sanoid/sanoid.defaults.conf *
+$current_user ALL=(ALL) NOPASSWD: /usr/bin/tee /etc/sanoid/*
+$current_user ALL=(ALL) NOPASSWD: /bin/chmod 440 /etc/sudoers.d/zfs-scripts
+$current_user ALL=(ALL) NOPASSWD: /bin/grep * /etc/sudoers.d/zfs-scripts"
+    
+    # Write the sudoers file using a temporary file for safety
+    local temp_file=$(mktemp)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)